### PR TITLE
SPLICE-1320 Add Support For Array Data Type

### PR DESCRIPTION
**Pull request by jleach4**
SPLICE-1320 Adding Array Type Support

### CODE REVIEW

**Comment by dgomezferro** [comment]
I created a crucible review in case anyone prefers it: https://codereview.splicedev.com:8443/cru/CR-676
I'll add my comments here though.

File: db-engine/src/main/java/com/splicemachine/db/iapi/types/DataTypeDescriptor.java [hunk 1]
```diff
@@ -788,6 +793,10 @@ public DataTypeDescriptor getDominantType(DataTypeDescriptor otherDTS,ClassFacto
             scale=higherType.getScale();
         }
 
+        lowerType.getChildren();
```
Comment (dgomezferro) [inline db-engine/src/main/java/com/splicemachine/db/iapi/types/DataTypeDescriptor.java]: Extra line

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/ArrayConstantNode.java [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+import java.util.List;
+
+/**
+ * This node represents an Array Initialization Node.  It supports arrays in the format
+ * [1,1,1] and will set them in the corresponding row to a SQLArray
+ *
+ *
+ */
+
+public class ArrayConstantNode extends ValueNode {
```
Comment (dgomezferro) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/ArrayConstantNode.java]: This class can be cleaned up a little bit, remove references to coalesce from comments if possible.

Comment (snowspeeder) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/ArrayConstantNode.java]: I suggest John could address these in next commit for minor cleanup, we need master back to green and this commit is reqd soon today, so I will merge it.

**Pull request merged**